content/docs/1.3.0/advanced-resources/os-distro-specific/upgrade-k8s-on-aks.md

```diff
@@ -0,0 +1,58 @@
+---
+title:  Upgrade Kubernetes on Azure AKS
+weight: 8
+---
+
+AKS with Longhorn can be upgraded by node-pool replacement. We cannot use direct `az aks upgrade` because the Longhorn data disk does not get migrated to the new OS image during the AKS upgrade re-image process.
```

Comment:
```suggestion
AKS provides `az aks upgrade` for in-places nodes upgrade by node reimaged, but however this will cause the original Longhorn disks missing, then there will be no disks allowing replica rebuilding in upgraded nodes anymore.

We suggest using node-pool replacement to upgrade the agent nodes but use `az aks upgrade` for control plane nodes to ensure data safety.
```

Reply:
Addressed.